src/main/java/frc/robot/RobotContainer.java

```diff
@@ -402,17 +404,24 @@ public void setAutoPlacementLEDs(Optional<Alliance> alliance) {
       subLEDs.setAnimationChunk(8, constLEDs.LED_NUMBER - 8, constLEDs.AUTO_ALIGNED_COLOR);
 
     } else {
-      subLEDs.setIndividualLED(rotationColor, 0);
-      subLEDs.setIndividualLED(rotationColor, 3);
-      subLEDs.setIndividualLED(rotationColor, 4);
-      subLEDs.setIndividualLED(rotationColor, 7);
-
-      subLEDs.setIndividualLED(XTranslationColor, 1);
-      subLEDs.setIndividualLED(XTranslationColor, 2);
-
-      subLEDs.setIndividualLED(YTranslationColor, 5);
-      subLEDs.setIndividualLED(YTranslationColor, 6);
-      subLEDs.clearAnimationChunk(8, constLEDs.LED_NUMBER - 8);
+      if (previousRotationColor != rotationColor) {
+        subLEDs.setIndividualLED(rotationColor, 0);
+        subLEDs.setIndividualLED(rotationColor, 3);
+        subLEDs.setIndividualLED(rotationColor, 4);
+        subLEDs.setIndividualLED(rotationColor, 7);
+        previousRotationColor = rotationColor;
+      }
+      if (previousXTranslationColor != XTranslationColor) {
+        subLEDs.setIndividualLED(XTranslationColor, 1);
+        subLEDs.setIndividualLED(XTranslationColor, 2);
+        previousXTranslationColor = XTranslationColor;
+      }
+      if (previousYTranslationColor != YTranslationColor) {
+        subLEDs.setIndividualLED(YTranslationColor, 5);
+        subLEDs.setIndividualLED(YTranslationColor, 6);
+        subLEDs.clearAnimationChunk(8, constLEDs.LED_NUMBER - 8);
```

Comment:
I think you'd want clearAnimationChunk() in all of the if statements, because what if only rotation is wrong but yTranslation stays the same? Our position is still wrong so we'd still want to clear the LEDs